classes/sushi/CounterR5Report.inc.php

```diff
@@ -0,0 +1,537 @@
+    /** The granularity of the usage data to include in the report. */
+    protected string $granularity = 'Month';
+
+    /**
+     * Get report name defined by COUNTER.
+     */
+    abstract public function getName(): string;
+
+    /**
+     * Get report ID defined by COUNTER.
+     */
+    abstract public function getID(): string;
+
+    /**
+     * Get report release.
+     */
+    public function getRelease(): string
+    {
+        return '5';
+    }
+
+    /**
+     * Get report description.
+     */
+    abstract public function getDescription(): string;
+
+    /**
+     * Get API path defined by COUNTER for this report.
+     */
+    abstract public function getAPIPath(): string;
+
+    /**
+     * Get request parameters supported by this report.
+     */
+    abstract public function getSupportedParams(): array;
+
+    /**
+     * Get filters supported by this report.
+     */
+    abstract public function getSupportedFilters(): array;
+
+    /**
+     * Get attributes supported by this report.
+     */
+    abstract public function getSupportedAttributes(): array;
+
+    /**
+     * Get used filters that will be displayed in the report header.
+     */
+    public function getFilters(): array
+    {
+        return $this->filters;
+    }
+
+    /**
+     * Get used attributes that will be displayed in the report header.
+     */
+    public function getAttributes(): array
+    {
+        return $this->attributes;
+    }
+
+    /**
+     * Set filters based on the requested parameters.
+     */
+    public function setFilters(array $filters): void
+    {
+        $this->filters = $filters;
+        foreach ($filters as $filter) {
+            switch ($filter['Name']) {
+                case 'Metric_Type':
+                    $this->metricTypes = explode('|', $filter['Value']);
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Set attributes based on the requested parameters.
+     */
+    public function setAttributes(array $attributes): void
+    {
+        $this->attributes = $attributes;
+        foreach ($attributes as $attribute) {
+            switch ($attribute['Name']) {
+                case 'Attributes_To_Show':
+                    $this->attributesToShow = explode('|', $attribute['Value']);
+                    break;
+                case 'granularity':
+                    $this->granularity = $attribute['Value'];
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Get report items
+     */
+    abstract public function getReportItems(): array;
+
+    protected function addWarning(array $exception): void
+    {
+        $this->warnings[] = $exception;
+    }
+
+    /**
+     * Process report parameters
+     *
+     * @throws SushiException
+     */
+    public function processReportParams($request, $params): void
+    {
+        $this->context = $request->getContext();
+
+        $this->checkRequiredParams($params);
+
+        $this->checkCustomerId($params);
```

Comment:
This relies on `$this->platformId`, but that doesn't get set until `$this->setPlatform` is called (just below). I get:
```
[20-Jul-2022 00:34:28 UTC] Slim Application Error:
Type: Error
Message: Typed property PKP\sushi\CounterR5Report::$platformId must not be accessed before initialization
File: /home/asmecher/git/ojs-main/lib/pkp/classes/sushi/CounterR5Report.inc.php
Line: 274
```